pkg/trace/api/otlp.go

```diff
@@ -398,6 +398,49 @@ func marshalEvents(events ptrace.SpanEventSlice) string {
 	return str.String()
 }
 
+// marshalLinks marshals span links into JSON.
+func marshalLinks(links ptrace.SpanLinkSlice) string {
+	var str strings.Builder
+	str.WriteString("[")
+	for i := 0; i < links.Len(); i++ {
+		l := links.At(i)
+		if i > 0 {
+			str.WriteString(",")
+		}
+		t := l.TraceID()
+		str.WriteString(`{"trace_id":"`)
+		str.WriteString(hex.EncodeToString(t[:]))
+		s := l.SpanID()
+		str.WriteString(`","span_id":"`)
+		str.WriteString(hex.EncodeToString(s[:]))
+		str.WriteString(`"`)
+		if l.Attributes().Len() > 0 {
+			str.WriteString(`,"attributes":{`)
```

Comment:
cc @songy23 
Why are we representing attributes as json object ? 
https://github.com/open-telemetry/opentelemetry-proto/blob/e6dfd7f4cefdc2e8844fa3dfcd8b307477d59937/opentelemetry/proto/trace/v1/trace.proto#L231-L234

We are using list in OTLP intake  https://github.com/DataDog/logs-backend/blob/a30e4d89db3e75ce73503ec0d280d580bb10a72e/domains/event-platform/libs/raw-content-parsers/src/main/java/com/fsmatic/workload/processing/SpansOTLPProtobufPayloadParser.java#L612

![image](https://user-images.githubusercontent.com/2471669/221678930-d22b11e0-19ce-431a-8e85-de89668da0db.png)

Reply:
Since the repeated key/value field in the proto specifies the keys to be unique, it seemed more natural to represent `attributes` as a map rather than a list of maps with one entry each (and it's consistent with the way the span entries represents attributes).

Reply:
As discussed offline, json object should be good.